Lib/fontTools/ttLib/tables/O_S_2f_2.py

```diff
@@ -508,11 +506,20 @@ def intersectUnicodeRanges(unicodes, inverse=False):
 	True
 	"""
 	unicodes = set(unicodes)
-	uniranges = _getUnicodeRangeSets()
-	bits = set([
-		bit for bit, unirange in enumerate(uniranges)
-		if not unirange.isdisjoint(unicodes) ^ inverse])
-	return bits
+	unicodestarts, unicodevalues = _getUnicodeRanges()
+	bits = set()
+	for code in unicodes:
+		stop, bit = unicodevalues[bisect.bisect(unicodestarts, code)]
+		if code <= stop:
+			bits.add(bit)
+	for code in unicodes:
+		# The spec says that bit 57 ("Non Plane 0") implies that there's
+		# at least one codepoint beyond the BMP; so I also include all
+		# the non-BMP codepoints here
+		if 0x10000 <= code < 0x110000:
+			bits.add(57)
+			break
+	return set(range(123)) - bits if inverse else bits
```

Comment:
I'm a little uneasy with the 123 literal. How about:
```python
   return set(range(len(OS2_UNICODE_RANGES))) - bits if inverse else bits
```

Reply:
Then I’ll change the test too :wink: